src/state/apis/swapper/helpers/validateTradeQuote.ts

```diff
@@ -216,6 +216,14 @@ export const validateTradeQuote = (
             const accountId =
               portfolioAccountIdByNumberByChainId[sellAssetAccountNumber][protocolFee.asset.chainId]
             const balanceCryptoBaseUnit = portfolioAccountBalancesBaseUnit[accountId][assetId]
+
+            if (firstHopSellFeeAsset?.assetId === assetId) {
+              return bnOrZero(balanceCryptoBaseUnit)
+                .minus(sellAmountCryptoBaseUnit)
+                .minus(protocolFee.amountCryptoBaseUnit)
+                .lt(0)
+            }
+
```

Comment:
The reason why it's high risk

We were never checking if the protocol asset require balance + it's the native asset and its also the asset we are selling, so we aren't taking in account that the balance has enough for the gas fees + protocol fees + sending amount

Another solution would be to add those amount to `sellAmountIncludingProtocolFeesCryptoBaseUnit` but it would err the code without any more messages so it would be less UX

Reply:
See https://github.com/shapeshift/web/pull/8267#discussion_r1871408061 - we probably want to make this Solana specific?